.buildkite/pipeline-helm-release-eck-resources.yml

```
@@ -0,0 +1,31 @@
+agents:
+  image: docker.io/library/golang:1.19.3
+  memory: "4G"
+
+steps:
+
+  - group: "release eck resources helm charts"
+    steps:
+
+      - label: "release eck-resources helm charts dev"
+        key: helm-release-eck-resources-dev
+        commands:
+          - cd hack/helm/release
+          # ignoring stdout to avoid "go: downloading ..." output.
+          - make build 2> /dev/null
+          - cp bin/releaser /tmp/releaser
+          # if a branch is sent as an environment variable, change to that branch.
+          - if [[ ! -z "$$HELM_BRANCH" ]]; then echo "checking out $$HELM_BRANCH branch" && git checkout $$HELM_BRANCH ; fi
+          - VAULT_ROOT_PATH="secret/ci/elastic-cloud-on-k8s" HELM_CREDENTIALS_FILE=/tmp/credentials.json /tmp/releaser --env=dev --charts-dir=../../../deploy --excludes=eck-operator --enable-vault
```

I think we can simplify a bit.
We can avoid the `cd` using `make -C`.
`VAULT_ROOT_PATH` isn't used, instead there is `HELM_VAULT_SECRET`. 
`HELM_CREDENTIALS_FILE` is set to `/tmp/credentials.jso` by default.
` --enable-vault` is true by default.
`--charts-dir` is set to `deploy` by default.

```suggestion
          - make -C hack/helm/release build 2> /dev/null
          - if [[ -n "$$HELM_BRANCH" ]]; then echo "checking out $$HELM_BRANCH branch" && git checkout $$HELM_BRANCH ; fi
          - hack/helm/release/bin/releaser --env=dev --excludes=eck-operator
```

---

Excellent point.  Adjusted all pipelines.